src/util/thunk-actions.js

```diff
@@ -23,13 +46,22 @@ export const createActionThunk = (type, func) => ({ ...props }) => async (
     else
       dispatch(action(thunkActionStatuses.EMPTY));
   } catch (error) {
+    if (isStaleAction({ cancelType, actionId }))
+      return;
+
     console.error(error);
     dispatch(action(thunkActionStatuses.ERROR));
   }
 };
 
-export const thunkActionStatuses = {
-  LOADING: 'LOADING',
-  EMPTY: 'EMPTY',
-  ERROR: 'ERROR'
+const generateActionId = () => Math.random();
+
+const isStaleAction = ({ cancelType, actionId }) =>
+  cancelType && actionStore[cancelType] !== actionId;
+
+export const cancelAction = ({ cancelTypeRegex }) => {
+  for (const key of Object.keys(actionStore)) {
```

Comment:
Is this `for` loop equivalent to: 
```
for (const key in actionStore) {
  ...
}
```
?

Reply:
Not exactly. `for...in` also returns keys that were inherited, whereas `for...of` with `Object.keys()` automatically includes a check that makes sure the keys are the ones that belong to that instance only. If you use `for...in`, eslint throws an error if you don't include `hasOwnProperty` checks, which are a pain to write correctly. Generally people recommend against using `for...in` these days.

Reply:
Good point. I forgot this difference.